### PR TITLE
Fix SDL2 gamma inconsistencies

### PR DESCRIPTION
**Pull request by jplatte**
Doesn't work yet, but opening a PR already to track progress.

### CODE REVIEW

**Comment by kvark** [comment]
Related - #1169

File: src/window/sdl/src/lib.rs [hunk 1]
```diff
@@ -12,14 +12,66 @@
...
+}
+
+impl From<WindowBuildError> for InitError {
+    fn from(e: WindowBuildError) -> Self {
+        InitError::WindowBuildError(e)
+    }
+}
+
+fn sdl2_pixel_format_from_gfx(format: Format) -> Option<PixelFormatEnum>
+{
```
Comment (kvark) [inline src/window/sdl/src/lib.rs]: nit: this shouldn't be on a new line

File: src/window/sdl/src/lib.rs [hunk 2]
```diff
@@ -28,27 +80,35 @@ use device_gl::Resources;
...
+    // TODO: Support different color channel types and/or other depth formats if possible
     use core::texture::{AaMode, Size};
 
-    let window = builder.opengl().build().unwrap(); //TODO: use actual settings
-    let context = window.gl_create_context().unwrap();
+    let mut window = builder.opengl().build()?;
+
+    let display_mode = DisplayMode {
+        format: sdl2_pixel_format_from_gfx(Cf::get_format())
+                    .ok_or(InitError::PixelFormatUnsupportedError)?,
```
Comment (kvark) [inline src/window/sdl/src/lib.rs]: the indentation is broken?

File: src/window/sdl/src/lib.rs [hunk 3]
```diff
@@ -28,27 +80,35 @@ use device_gl::Resources;
...
 /// }
 /// ```
-pub fn init(builder: &mut sdl2::video::WindowBuilder) ->
-    (sdl2::video::Window, sdl2::video::GLContext,
-     device_gl::Device, device_gl::Factory,
-     handle::RenderTargetView<Resources, Srgba8>,
-     handle::DepthStencilView<Resources, DepthStencil>)
+pub fn init<Cf>(builder: &mut WindowBuilder) -> Result<InitOk<Cf, DepthStencil>, InitError>
+where
+    Cf: RenderFormat<Channel = Srgb>,
```
Comment (kvark) [inline src/window/sdl/src/lib.rs]: it seems like a required limitation

File: src/window/sdl/src/lib.rs [hunk 4]
```diff
@@ -28,27 +80,35 @@ use device_gl::Resources;
...
-    let context = window.gl_create_context().unwrap();
+    let mut window = builder.opengl().build()?;
+
+    let display_mode = DisplayMode {
+        format: sdl2_pixel_format_from_gfx(Cf::get_format())
+                    .ok_or(InitError::PixelFormatUnsupportedError)?,
+        ..window.display_mode()?
+    };
+    window.set_display_mode((Some(display_mode)))?;
+    window.subsystem().gl_attr().set_framebuffer_srgb_compatible(true);
```
Comment (kvark) [inline src/window/sdl/src/lib.rs]: we should only pass `true` there if the channel type is sRGB

File: src/window/sdl/src/lib.rs (hunk 2 above)
Comment (jplatte) [inline src/window/sdl/src/lib.rs]: Well this was intentional, I indented 4 spaces past the start of the expression for initializing the format field. But I'm perfectly fine with just running rustfmt over everything (even though its rules are probably gonna change soon[-ish]).

**Comment by kvark** [comment]
Closing in favour of #1173

**Pull request closed without merging**